stacker/dag/__init__.py

```diff
@@ -414,14 +414,12 @@ class ThreadedWalker(object):
     allows, using threads.
 
     Args:
-        semaphore (threading.Semaphore, optional): a semaphore object which
+        semaphore (threading.Semaphore): a semaphore object which
```

Comment:
The type annotation here isn't _quite_ accurate, since the default case of `UnlimitedSemaphore` isn't a `threading.Semaphore`, but I felt like this was fine.